src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -89,13 +91,17 @@ private static boolean isValidPath(Path filePath) {
     /**
      * Saves all data to this storage file.
      *
-     * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws StorageOperationException if there were errors finding storage file, converting and/or storing data to file.
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
 
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+		File storageFile = new File(DEFAULT_STORAGE_FILEPATH);
+		if (!(storageFile.exists())) { 
+			throw new StorageOperationException("Error finding file: " + DEFAULT_STORAGE_FILEPATH);
```

Comment:
You have thrown the exception in the right place. But where are you handling it?

The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception